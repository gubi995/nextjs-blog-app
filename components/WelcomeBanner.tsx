import React from 'react';
import { Avatar, Typography, styled } from '@material-ui/core';

const TypographyContrast = styled(Typography)(({ theme }) => ({ color: theme.palette.primary.contrastText }));

const StyledWelcomeBanner = styled('div')(({ theme }) => ({
  height: '400px',
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  textAlign: 'center',
  backgroundColor: theme.palette.primary.main,
  clipPath: 'polygon(0% 0%, 100% 0, 100% 70%, 50% 100%, 0 70%)',
}));

const StyledAvatar = styled(Avatar)(({ theme }) => ({ height: '140px', width: '140px', margin: theme.spacing(3) }));

const WelcomeBanner = () => {
  return (
    <StyledWelcomeBanner>
      <StyledAvatar src="/itsme.jpg" />
      <TypographyContrast variant="h4">Welcome here my friend</TypographyContrast>
      <TypographyContrast variant="subtitle1">Check out my repos and blog posts</TypographyContrast>
    </StyledWelcomeBanner>
  );
};

export default WelcomeBanner;
