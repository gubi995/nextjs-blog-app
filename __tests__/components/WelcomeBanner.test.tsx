import React from 'react';
import { render } from '@testing-library/react';

import WelcomeBanner from '../../components/WelcomeBanner';

describe('WelcomeBanner', () => {
  test("should rendered component matches with it's the snapshot", () => {
    const { asFragment } = render(<WelcomeBanner />);

    expect(asFragment()).toMatchSnapshot();
  });
});
