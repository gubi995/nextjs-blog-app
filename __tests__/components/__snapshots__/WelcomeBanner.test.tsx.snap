// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeBanner should rendered component matches with it's the snapshot 1`] = `
<DocumentFragment>
  <div
    class="div-root-1 div-root-2"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circle WithStyles(ForwardRef(Avatar))-root-3 WithStyles(ForwardRef(Avatar))-root-4"
    >
      <img
        class="MuiAvatar-img"
        src="/itsme.jpg"
      />
    </div>
    <h4
      class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-12 WithStyles(ForwardRef(Typography))-root-13 MuiTypography-h4"
    >
      Welcome here my friend
    </h4>
    <h6
      class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-12 WithStyles(ForwardRef(Typography))-root-44 MuiTypography-subtitle1"
    >
      Check out my repos and blog posts
    </h6>
  </div>
</DocumentFragment>
`;
